mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -284,8 +285,9 @@ public void close() throws IOException {
       boolean applyResidual = !context.getConfiguration().getBoolean(InputFormatConfig.SKIP_RESIDUAL_FILTERING, false);
 
       if (applyResidual && residual != null && residual != Expressions.alwaysTrue()) {
+        InternalRecordWrapper wrapper = new InternalRecordWrapper(readSchema.asStruct());
```

Comment:
Can we add a comment here why this wrapper is needed? I think the class name doesn't tell you much and it does not have javadoc either to explain its purpose. What do you think?

Reply:
Added comment